src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.AnalyzerExecutionContext.cs

```diff
@@ -173,6 +173,16 @@ void processMembers(IEnumerable<ISymbol> members)
                             {
                                 memberSet = memberSet ?? new HashSet<ISymbol>();
                                 memberSet.Add(member);
+
+                                // Ensure that we include symbols for both parts of partial methods.
+                                if (member is IMethodSymbol method)
+                                {
+                                    var otherPartOfPartialOpt = method.PartialDefinitionPart ?? method.PartialImplementationPart;
```

Comment:
>method.PartialDefinitionPart ?? method.PartialImplementationPart [](start = 64, length = 64)

Why are we trying both? Don't we know which symbol is going to be in the members list when both declarations are present? I assume that would be the definition part.

Reply:
Personally, I wasn't sure what part is returned by `GetMembers()` invocation and was just playing safe. If you are certain that `GetMembers()` always returns the definition part (and that cannot change in future), then I can change the above code to just add `PartialImplementationPart`, if non-null.

Reply:
I see we do remove the implementation part [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/Source/SourceMemberContainerSymbol.cs,2529), safe to assume that will always be guaranteed?